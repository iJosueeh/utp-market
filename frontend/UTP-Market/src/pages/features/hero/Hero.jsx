import React from 'react';
import image1 from "../../../assets/image1.webp";
const Hero = () => {
    return (
 <section className="container py-5"style={{ marginTop: "200px",marginBottom: "200px"}}>
      <div className="row align-items-center">
        <div className="col-md-5 text-center mb-4 mb-md-0">
          <img
            src={image1}
            alt="UTP Market"
            className="img-fluid rounded"
          />
        </div>

        <div className="col-md-6 ps-md-5">
          <h6 className="text-danger fw-bold">UTP - MARKET</h6>
          <h1 className="fw-bold text-dark">
            Tu marketplace confiable hecho por y para estudiantes UTP
          </h1>
          <p className="text-muted fs-5 mt-3">
            Cada vez más estudiantes usan nuestra página para comprar y vender
            servicios, materiales, guías y otros recursos académicos.
          </p>

          
          <div className="d-flex gap-3 mt-4">
            <button className="btn btn-danger px-4 py-2">
              Explorar productos
            </button>
            <button className="btn btn-outline-danger px-4 py-2">
              Únete a la comunidad
            </button>
          </div>
        </div>
      </div>
    </section>
  );
};


export default Hero;