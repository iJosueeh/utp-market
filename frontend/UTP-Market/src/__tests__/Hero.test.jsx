import React from 'react';
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Hero from "../pages/features/hero/Hero";

describe("Hero component", () => {
  it("debe coincidir con el snapshot", () => {
    const { container } = render(<Hero />);
    expect(container).toMatchSnapshot();
  });

  it("renderiza la imagen con el alt correcto", () => {
    render(<Hero />);
    const image = screen.getByAltText(/UTP Market/i);
    expect(image).toBeInTheDocument();
  });

  it("muestra el subtítulo 'UTP - MARKET'", () => {
    render(<Hero />);
    expect(screen.getByText(/UTP - MARKET/i)).toBeInTheDocument();
  });

  it("muestra el título principal", () => {
    render(<Hero />);
    expect(
      screen.getByText(/Tu marketplace confiable hecho por y para estudiantes UTP/i)
    ).toBeInTheDocument();
  });

  it("muestra la descripción", () => {
    render(<Hero />);
    expect(
      screen.getByText(/Cada vez más estudiantes usan nuestra página/i)
    ).toBeInTheDocument();
  });

  it("renderiza los botones correctamente", () => {
    render(<Hero />);
    expect(
      screen.getByRole("button", { name: /Explorar productos/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /Únete a la comunidad/i })
    ).toBeInTheDocument();
  });
});
