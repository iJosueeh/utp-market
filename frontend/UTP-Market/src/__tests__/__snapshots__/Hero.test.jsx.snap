// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero component > debe coincidir con el snapshot 1`] = `
<div>
  <section
    class="container py-5"
    style="margin-top: 200px; margin-bottom: 200px;"
  >
    <div
      class="row align-items-center"
    >
      <div
        class="col-md-6 text-center mb-4 mb-md-0"
      >
        <img
          alt="UTP Market"
          class="rounded-circle img-fluid shadow"
          src="/src/assets/image1.webp"
          style="width: 550px; height: 450px; object-fit: cover;"
        />
      </div>
      <div
        class="col-md-6"
      >
        <h6
          class="text-danger fw-bold"
        >
          UTP - MARKET
        </h6>
        <h1
          class="fw-bold text-dark"
        >
          Tu marketplace confiable hecho por y para estudiantes UTP
        </h1>
        <p
          class="text-muted fs-5 mt-3"
        >
          Cada vez más estudiantes usan nuestra página para comprar y vender servicios, materiales, guías y otros recursos académicos.
        </p>
        <div
          class="d-flex gap-3 mt-4"
        >
          <button
            class="btn btn-danger px-4 py-2"
          >
            Explorar productos
          </button>
          <button
            class="btn btn-outline-danger px-4 py-2"
          >
            Únete a la comunidad
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
